### PR TITLE
Project specification

### PR DESCRIPTION
**Pull request by Scarletsang**
A HTTP request object needs to be produced by a parser.
Also, after the body generation is completed, a HTTP response has to wrap the body to return back to the socket manager/ write directly to the output stream.

Currently only the grammers are added for the parser, so that I understand what is needed for the HTTP request object. Not all the grammers here will be implemented.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,33 @@ http {
...
+Body fields
+
+- Content-Type
+- Content-Length
+- Content-Location
+
+Request fields
+
+- Host
+- Referer
```
[the remark this answers is not in the document]
Comment (Scarletsang) [reply README.md]: - Host contains the domain name. It is needed to match the correct server blocks in the configuration file
- Referer contains the url that the client uses (excluded fragments and userinfo components of the URI reference), it represents who the client is. If the client is a malicious website, the server can see the Referer header and chooses to decline the request with a 404.

File: README.md [hunk 2]
```diff
@@ -79,12 +79,12 @@ http {
         }
         location /images {
             allow GET;
+			redirect /new;
```
[the remark this answers is not in the document]
Comment (Scarletsang) [reply README.md:72]: Yes I will change the specification

**Comment by Scarletsang** [comment]
This pull request is now just to specify the scope of the project. What we handle and what we don't handle in the configuration file (that represents the functionality of the program) and the HTTP message headers

**Review by Sangyao42 — APPROVED** [review]
Approved the redirection

**Review by euns-kim — APPROVED** [review]

**Comment by Sangyao42** [comment]
I need only unique ports from config file.
`server{
 listen 80
 server_name my.com
}
server{
 listen 80
server_name your.com
}
`
These two server should share the same server socket. In Configuration Class, give me a getter for getting unique ports.

**Pull request merged**